homeassistant/components/mqtt/vacuum/schema_state.py

```diff
@@ -218,7 +242,7 @@ def state_message_received(msg):
             self._update_state_attributes(payload)
             get_mqtt_data(self.hass).state_write_requests.write_state_request(self)
 
-        if self._config.get(CONF_STATE_TOPIC):
+        if self._config.get(CONF_STATE_TOPIC) is not None:
```

Comment:
Slight code change here. is it on purpose?
And could you maybe use a walrus to avoid readying ig again two lines down?
```suggestion
        if (state_topic := self._config.get(CONF_STATE_TOPIC)) is not None:
```
or
```suggestion
        if state_topic := self._config.get(CONF_STATE_TOPIC):
```